### ISSUE TITLE
Live aggregation creates table even though table is not used

### ISSUE THREAD

**Issue by wastaz** [JasperFx/marten#1757]
### To  reproduce:
```
// Configure marten to allow it to create tables
// Create a stream with some events

var aggregate = await session.Events.AggregateStreamAsync<MyProjection>(streamId);
```

### Expected:
Marten reads event stream from database to memory and creates an aggregate of the expected type with all events applied.

### Actual:
Marten reads event stream from database to memory and creates an aggregate of the expected type with all events applied.
*AND*
Marten creates an empty mt_doc_myprojection table in the database. (If store options allows creating tables.)

Marten version: 3.13.3
Running on: windows, net5.0.102

Id understand creating the table if calling a live projection would also save a cached version of it in the table at that time. But it doesnt do that automatically (which I also think is good, less magic, Id be surprised if aggregatestreamasync would do that).

**Comment by jeremydmiller** [comment]
Just verified that the V4 code in master does *not* do this. I'd actually like to punt this, get v4 released, and say "won't fix, please upgrade"

**Comment by jeremydmiller** [comment]
I'm only closing this because I just don't want to have to run into the 3.* code if that's okay @wastaz. Re-open this if you want it fixed before 4.0.
